test/Tests/Web/RequestHandlerTests.cs

```diff
@@ -40,6 +41,23 @@ public class RequestHandlerTests : TestBase
             response.Warnings[0].ToString().ShouldEqual("GraphQL.Validation.ValidationError: Variable \"$foo\" is never used in operation \"$test\".");
         }
 
+        [Test]
+        public async Task Can_Run_Query_With_ComplexityConfiguration()
+        {
+            var request = Request.New("{ \"query\": \"{ hello }\" }");
+            var response = await RequestHandler
+                .New()
+                .WithQuery<TestQuery>()
+                .WithComplexityConfiguration(new ComplexityConfiguration() {MaxDepth = 2})
+                .Generate()
+                .ProcessRequest(request, null);
+
+            response.ExecutionResult.Data.ShouldHaveFieldWithValue("hello", "World");
+            response.Body.ShouldEqual("{\"data\":{\"hello\":\"World\"}}");
+            response.Errors.Count.ShouldEqual(0);
+            response.Warnings.Count.ShouldEqual(0);
+        }
+        
         class TestQuery
```

Comment:
Maybe also add an additional test that is too complex according to given `MaxDepth`. Only then you're sure that the `ComplexityConfiguration` is actually being used.